pkg/apis/options/legacy_options.go

```diff
@@ -509,6 +509,10 @@ type LegacyProvider struct {
 	UserIDClaim                        string   `flag:"user-id-claim" cfg:"user_id_claim"`
 	AllowedGroups                      []string `flag:"allowed-group" cfg:"allowed_groups"`
 
+	SkipAudCheckWhenMissing       bool     `flag:"skip-aud-check-when-missing" cfg:"skip_aud_check_when_missing"`
+	AudienceVerificationClaim     string   `flag:"audience-verification-claim" cfg:"audience_verification_claim"`
+	ExtraAudiencesForVerification []string `flag:"extra-audiences-for-verification" cfg:"extra_audiences_for_verification"`
```

Comment:
Can we split this functionality out from this PR?

And open an Issue to discuss? I'm not sure this is something we want to support I"m not sure I like freeform audiences that aren't explicitly attached to an issuer that matches it.

`extra-jwt-issuers` will likely be replaced by a concept of `BearerProviders` when we have structured configurations. Where any provider that implements the `CreateSessionFromToken` provider interface method can plug in as a Bearer handler.

At the moment people get super confused conflating these `extra-jwt-issuer` default OIDC handling with their main OIDC provider settings.